roles/ovirt-datacenter-cleanup/tasks/vms.yml

```diff
@@ -3,11 +3,13 @@
     auth: "{{ ovirt_auth }}"
     pattern: "datacenter={{ data_center_name }}"
 
-- name: Remove VMs
+- name: Remove all VMs apart from HE one if exists
   ovirt_vms:
     state: absent
     id: "{{ ovirt_item.id }}"
     auth: "{{ ovirt_auth }}"
   with_items: "{{ ovirt_vms }}"
+  when: not hosted_engine or
```

Comment:
Hmm, this works? Shouldn't it be:

```yaml
when:
  - not hosted_engine_remove or (hosted_engine_remove and ovirt_item.name != hosted_engine_vm_name)
```

Reply:
Works fine, and it's in docs as well.
check here second example http://docs.ansible.com/ansible/latest/playbooks_conditionals.html#the-when-statement